src/http/RequestMethod.h

```diff
@@ -28,6 +28,7 @@ class HttpRequestMethod : public RefCountable
 public:
     HttpRequestMethod() : theMethod(Http::METHOD_NONE), theImage() {}
     HttpRequestMethod(Http::MethodType const aMethod) : theMethod(aMethod), theImage() {}
+    HttpRequestMethod(const HttpRequestMethod &) = default;
```

Comment:
The combination of a defaulted copy constructor for X and a customized X assignment operator is a red flag for many readers. Most likely, you want to default both (in this case).

Reply:
The customized assignment operator is like the default. I suppose it was written to also support assigning from Http::MethodType via operator overload, but the implementation is just like the default one so I'm changing it to that.

Reply:
> Alex: The combination of a defaulted copy constructor for X and a customized X assignment operator is a red flag for many readers

Actually, in this specific case, it is not just a red flag but a big bug AFAICT. The defaulted constructor copies the RefCountable::count_ from another object and that will leak this reference counting object (if it is ever used as a reference counted object).

> Alex: Most likely, you want to default both (in this case).

I guessed incorrectly. I bet I did not notice `public RefCountable`. This tells me that our Lock declaration is itself buggy because it let you get away with this implementation bug. Lock should prohibit such copying but does not.

What scares me is that prohibiting Lock copying in master leads in many compilation errors. That means that we already have similar bugs in Squid code! This implies that the Lock fix deserves a dedicated PR. Please submit such a PR if you can. Needless to say, this PR must not make things worse and currently does.